torch/optim/sgd.py

```diff
@@ -334,7 +334,14 @@ def _single_tensor_sgd(
         grad = grads[i] if not maximize else -grads[i]
 
         if weight_decay != 0:
-            grad = grad.add(param, alpha=weight_decay)
+            # Nested if is necessary to bypass jitscript rules
+            if isinstance(weight_decay, Tensor):
+                if weight_decay.requires_grad:
```

Comment:
can you add a comment here saying

# used for the differentiable path, which is why the param.clone() is needed